helpers/lagoonenvclean.sh

```diff
@@ -0,0 +1,182 @@
+	LAGOONENVS[$1,envs]="$ENVS"
+}
+
+# Function to retrieve information about single project's environment
+environment_query() {
+	
+	# Escape environment name
+	ENVCLEAN=$(echo $1 | sed -e "s#/#\\\/#g")
+
+	# Substitute and escape characters in GraphQL query
+	ONEENVQUERY=$(echo $QLENVQUERY | sed "s/NAME/\"$ENVCLEAN\"/g" | sed "s/ID/$2/g" | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+
+	# Run the query and store results into a variable
+	ONERESULT=`curl -s -k -X POST -H 'Content-Type: application/json' -H "$AMZBEARERTKN" $ENDPOINT -d "{\"query\": \"$ONEENVQUERY\"}"`
+
+	# Set variable with the last update of an environment, in a YYYYMMDD format
+	UPDATEDATE=`echo "$ONERESULT" | jq -r '.[].environmentByName.updated' | cut -f1 -d " "| xargs date +%Y%m%d -d`
+
+	# Declare a "fake" multi-demensional array, where to store variable per project and per environment
+	declare -gA LAGOONENV
+	LAGOONENV[$i,$1]="$UPDATEDATE"
+	#echo "Project $i, env $1, last update ${LAGOONENV[$i,$1]}"
+}
+
+
+# Function to delete Lagoon project's environment
+lagoonenv_clean() {
+
+	# Substitue and escape characters in GraphQL query
+	ENVDELETE=$(echo $QLENVDELETE | sed "s/ENV/\"$1\"/g" | sed "s/PROJECT/\"$2\"/g" | sed "s/BOOLEAN/\"$3\"/g" | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+	if [ "$3" = "false" ]; then
+		echo "Dry-run env delete"
+		echo $ENVDELETE
+	else
+		echo "Deleting the $1 env of $2 project"
+		echo $ENVDELETE
+	fi
+}
+
+# Function to query GitHub PR's status
+ghpr_query() {
+
+	# Retrieve the status of a PR
+	GHPRSTATE=$(curl -s -k -H "$GHBEARERTKN" $GHENDPOINT/repos/$1/$2/pulls/$3|jq -r .state)
+
+	# Retrieve the lastupdate time of a PR
+	GHPRUPDATE=$(curl -s -k -H "$GHBEARERTKN" $GHENDPOINT/repos/$1/$2/pulls/$3|jq -r .updated_at| xargs date +%Y%m%d -d)
+	echo "PR n$3 is $GHPRSTATE and updated on $GHPRUPDATE"
+
+	# Execute the clean function *only* if the PR is closed and the lagoon environment last update time is before last N months.
+	if [ "$GHPRSTATE" = "closed" -a $GHPRUPDATE -le $CLEANDATE ]; then
+		echo "Delete environment ${env} $DRYRUN"
+		lagoonenv_clean $env $i $DRYRUN
+	else
+		if [ "$GHPRSTATE" = "open" ]; then
+			echo "Environment ${env} is not deleted because status"
+		else
+			echo "Environment ${env} is not deleted because date"
+		fi
+	fi
+}
+
+usage() {
+	echo -e "Usage is: ./lagoonenvclean.sh customer number_of_months true|false \nes: ./lagoonenvclean.sh rasch 4 true\n"
+	echo "Customer parameter is mandatory!"
+	exit 1
+}
+
+
+# Main function
+main () {
+	if [ -z "$1" ]; then
+		usage
+	fi
+	# Script body
+	if [ -z "$AMZTKN" -o -z "$GHUTKN" ]; then
+		echo "Lagoon JWT token or Github Token are not set"
+		exit 1
+	fi
+	allproject_query $1
+	for i in $PROJECTS
+	do
+		allenvironment_query $i
+		for env in ${LAGOONENVS[$i,envs]}
+		do
+			id=${LAGOONENVS[$i,id]}
+			environment_query "$env" $id
```

Comment:
the result of `environment_query` does not seem to be used anywhere?

Reply:
Removed